mnemonic-core/src/main/java/org/apache/mnemonic/MemStream.java

```diff
@@ -17,51 +17,139 @@
 
 package org.apache.mnemonic;
 
-import java.io.File;
-
 /**
- * a memory file that manages its data on native memory storage. Note: this
- * class depends on PMFS, we suggest that NVM library to support this feature in
- * native layer. In addition, the permission of /mnt/pmfs will be set properly.
- * 
+ * This interface to randomly access the memory. 
  *
  */
-public class MemStream extends File {
+public interface MemStream {
+
+  /**
+   * This function will close the random access mem stream and release any 
+   * system resources associated with the stream.
+   * 
+   */
+  void close(); 
 
-  private static final long serialVersionUID = 6579668848729471173L;
-  private String uri, id;
+  /**
+   * This function returns the MemHolder associated with this mem stream
+   * Not sure how to return different kinds of MemHolder
+   * Here use the abstract class MemHolder
+   * 
+   * @return MemHolder
+   */
+  MemHolder getMem(); 
```

Comment:
Please remove it because we don't expect to expose the internal structure to user.